llama_index/agent/openai_agent.py

```diff
@@ -53,6 +70,71 @@ def call_function(
     )
 
 
+class StreamingChatResponse:
+    """Streaming chat response to user and writing to chat history."""
+
+    def __init__(
+        self, chat_stream: Union[ChatResponseGen, ChatResponseAsyncGen]
+    ) -> None:
+        self.chat_stream = chat_stream
+        self.queue: queue.Queue = queue.Queue()
+        self.is_done = False
+        self.is_function: Optional[bool] = None
+        self.response = ""
```

Comment:
very nit: could we "hide" these with leading underscore e.g. `self._is_done` if they are not meant to be user facing?

Reply:
helps to understand how caller interact with this object.